packages/lesswrong/components/comments/DebateResponse.tsx

```diff
@@ -0,0 +1,222 @@
+    default:
+      return 'yellowBorder';
+  }
+};
+
+export const DebateResponse = ({classes, comment, replies, idx, responseCount, orderedParticipantList, responses, post}: {
+  classes: ClassesType,
+  comment: CommentsList,
+  replies: CommentsList[],
+  idx: number,
+  responseCount: number,
+  orderedParticipantList: string[],
+  responses: DebateResponseWithReplies[],
+  post: PostsWithNavigation | PostsWithNavigationAndRevision,
+}) => {
+    const { captureEvent } = useTracking(); //it is virtuous to add analytics tracking to new components
+    const { CommentUserName, CommentsItemDate, CommentBody, CommentsEditForm, CommentsMenu, DebateCommentsListSection } = Components;
+
+    const responseStates = responses.map(_ => false);
+    const [showReplyState, setShowReplyState] = useState([...responseStates]);
+    const [showEdit, setShowEdit] = useState([...responseStates]);
+
+    const fullParticipantSet = new Set([post.userId, ...(post.coauthorStatuses ?? []).map(coauthor => coauthor.userId)]);
+
+    const showRepliesForComment = (e: React.MouseEvent, responseIdx: number) => {
+      e.preventDefault();
+      const newReplyState = [...showReplyState];
+      newReplyState[responseIdx] = !newReplyState[responseIdx];
+      setShowReplyState(newReplyState);
+    };
+
+    const showEditForResponse = (newShowEditState: boolean, responseIdx: number) => {
+      const newShowEdit = [...showEdit];
+      newShowEdit[responseIdx] = newShowEditState;
+      setShowEdit(newShowEdit);
+    }
+
+    const currentUser = useCurrentUser();
+
+    const isFirstCommentInBlock = idx === 0;
+    const isLastCommentInBlock = idx === (responseCount - 1);
+    const commentParticipantIndex = orderedParticipantList.indexOf(comment.userId);
+    const readerIsParticipant = currentUser && fullParticipantSet.has(currentUser._id);
+
+    const showHeader = isFirstCommentInBlock;
+    const showInlineReplyForm = isLastCommentInBlock && !readerIsParticipant;
+    const showReplyLink = replies.length > 0 || showInlineReplyForm;
+    const addBottomMargin = isLastCommentInBlock;
+    const borderStyle = getParticipantBorderStyle(commentParticipantIndex);
+
+    const header = showHeader && <>
+      <CommentUserName comment={comment} className={classes.username} />
+      <span>{" " /* Explicit space (rather than just padding/margin) for copy-paste purposes */}</span>
+      <CommentsItemDate comment={comment} post={post} />
+    </>;
+
+    const menu = <CommentsMenu
+      comment={comment}
+      post={post}
+      showEdit={() => showEditForResponse(true, idx)}
+      className={classes.menu}
+    />;
+
+    const votingSystemName = comment.votingSystem || "default";
+    const votingSystem = getVotingSystemByName(votingSystemName);
+    const voteProps = useVote(comment, "Comments", votingSystem);
+    const commentItemRef = useRef<HTMLDivElement|null>(null); // passed into CommentsItemBody for use in InlineReactSelectionWrapper
```

Comment:
For consistency's sake, I'd pull these up to the top of the component logic to live with the other hooks